### PR TITLE
Add Julia test workflow

### PR DESCRIPTION
**Pull request by etpinard**
As discussed in https://github.com/plotly/Dash.jl/pull/175#issuecomment-1327524286, I think it would be easier to run the Julia `] test` for different Julia versions inside a Github workflow instead of inside the [`plotly/julia:ci`](https://hub.docker.com/r/plotly/julia) docker container. 

Running Julia tests inside Github workflows is fairly easy now thanks to the [julia-actions](https://github.com/julia-actions) organisation. This should make it much easier to maintain the list of tested Julia versions for Dash.jl. It's much easier to patch a yaml file than updating a docker image, right? :stuck_out_tongue: 

I'm thinking we could also drop these lines:

https://github.com/plotly/Dash.jl/blob/f0606e07d2479fd8b5be1da4a6a59656e24acfa3/.circleci/config.yml#L30-L33

https://github.com/plotly/Dash.jl/blob/f0606e07d2479fd8b5be1da4a6a59656e24acfa3/test/ci_prepare.jl#L11

and have the CircleCI workflow be used just for the integration (percy) tests.

### CODE REVIEW

File: .github/workflows/jl_test.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        jl_version: ["1.3", "1.6", "1.8"]
+        jl_version: ["1.4", "1.6", "1.8"]
```
Comment (etpinard) [inline .github/workflows/jl_test.yml:11]: Testing on Julia 1.3 leads to 

image

We should probably update

https://github.com/plotly/Dash.jl/blob/f0606e07d2479fd8b5be1da4a6a59656e24acfa3/Project.toml#L39

to match the [PlotlyBase.jl compat](https://github.com/sglyon/PlotlyBase.jl/blob/72cfaf977edaaa9f51c7264810c6e44ed4864bb2/Project.toml#L27)

**Comment by etpinard** [comment]
Hmm, I'll need to figure out why the final step of the CircleCI workflow is failing

image

**Comment by mind6** [comment]
> As discussed in [#175 (comment)](https://github.com/plotly/Dash.jl/pull/175#issuecomment-1327524286), I think it would be easier to run the Julia `] test` for different Julia versions inside a Github workflow instead of inside the [`plotly/julia:ci`](https://hub.docker.com/r/plotly/julia) docker container.
> 
> Running Julia tests inside Github workflows is fairly easy now thanks to the [julia-actions](https://github.com/julia-actions) organisation. This should make it much easier to maintain the list of tested Julia versions for Dash.jl. It's much easier to patch a yaml file than updating a docker image, right? 😛
> 
> I'm thinking we could also drop these lines:
> 
> https://github.com/plotly/Dash.jl/blob/f0606e07d2479fd8b5be1da4a6a59656e24acfa3/.circleci/config.yml#L30-L33
> 
> https://github.com/plotly/Dash.jl/blob/f0606e07d2479fd8b5be1da4a6a59656e24acfa3/test/ci_prepare.jl#L11
> 
> and have the CircleCI workflow be used just for the integration (percy) tests.

The github actions seem quite nice. Don't know anything about CircleCI/percy hope it works out.

**Comment by etpinard** [comment]
We have passing tests :tada: - merging!

**Pull request merged**